custom_components/better_thermostat/helpers.py

```diff
@@ -252,6 +253,9 @@ async def startup(self):
 				self.name
 			)
 			self._hvac_mode = HVAC_MODE_OFF
+		self._bt_hvac_mode = self._hvac_mode
+		self._last_window_state = self.window_open
+		self._cur_temp = float(sensor_state.state)
```

Comment:
Please don't cast unprotected here. The sensor might contain "unavailable", "unknown" or is just None.

Use models.utils.convert_to_float() for this. :)